### ISSUE TITLE
possible false positive for "Form element has multiple <label> elements"

### ISSUE THREAD

**Issue by mfairchild365** [dequelabs/axe-core#202]
Consider the following example:

> Two labels exist for the same form element, but only one of which is ever displayed at a time. Which label is displayed depends on the screen width (responsive design), the other label is hidden via `display:none`. The labels are identical, just displayed at different locations for mobile optimization.

Am I correct in assuming that in the above scenario, it is okay to have two labels? Right now AXE will detect both regardless of visibility.

I'm also curious about the rationale behind this rule. The HTML standard allows a one-to-many relationship (a control can have multiple labels). My understanding is that if a screen reader does not support this, it is more of a screen reader bug than an accessibility bug. While it is important to to understand and account for the fact that screen readers might not implement the standards correctly, it seems rather extreme to implement an automated rule for it (rather it should be addressed as a screen reader bug, but that is more of an ideology issue). I've done a bit of digging but haven't found much in the way of recent testing for screen reader support of multiple label elements.

I discovered this while testing using the AXE extension in google chrome, version 2.0.0

**Comment by dylanb** [comment]
@mfairchild365 Thanks for the feedback!

I see two separate issues:

1) Duplicate label that is hidden should not trigger this rule - I agree  this is a bug. We will use this issue to track this - thank you for pointing this out.

2) Rationale behind flagging this in the first place: WCAG 2 has a requirement for techniques to be used they must be what is called ["accessibility supported"](https://www.w3.org/TR/UNDERSTANDING-WCAG20/conformance.html#uc-accessibility-support-head), when you boil all of that down it comes down to simply "they must work". Because the double label does not work, we flag it to point out to the developer that they are using a technique that will not achieve what they want if this was their intent (but more often this is the result of code being copied and the ID not being changed)

@marcysutton can you create a ticket in JIRA to track this?

**Comment by marcysutton** [comment]
Yep will do!

**Comment by mfairchild365** [comment]
@dylanb Thank you for the quick response.

2) that is a very good point. I was curious as to if this is still an issue and was able to confirm with VO.

1) While testing with VO, I was able to confirm that only the first label is associated with the input. The second label is not. So if the first label has `display:none`, VO thinks that the input has no label.

This is the code that I tested with
```
<label for="test-input" style="display:none">label one</label>
<label for="test-input">label two</label>
<input id="test-input" type="text">
```

So, with respect to issue 2 ("accessibility supported"), I think that the current test is accurate and this is not actually an AXE bug.

**Comment by marcysutton** [comment]
That's really interesting. I wonder if it works better to put a span inside of the label and use media-query-based `display: none` on that instead.

**Comment by dylanb** [comment]
Anything that is `display:none` will not work as a label unless using `aria-labelledby`. So we should change the rule to NOT trigger if the second instance is `display:none` but the first one isn't. So the following should pass

```
		<label for="test-input2" >label one</label>
		<label for="test-input2" style="display:none">label two: hidden because the first one is display:none</label>
		<input id="test-input2" type="text">
```

**Comment by marcysutton** [comment]
I've fixed this internally, it will be available in our next axe-core release.

**Comment by marcysutton** [comment]
Fixed with https://github.com/dequelabs/axe-core/commit/2324e2a3405573e1bf2e50d6f1eb90f9214ad739. Stay tuned for axe-core 2.2.0.
